### PR TITLE
Nordic Countries

### PR DESCRIPTION
**Pull request by dmichalowicz**
Calendars for Denmark, Finland, Norway and Sweden.

### CODE REVIEW

File: trading_calendars/exchange_calendar_xosl.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    Early Closes:
+      - Holy Wednesday
+    """
+    name = 'XOSL'
+    tz = timezone('Europe/Oslo')
+    open_times = (
+        (None, time(9, 1)),
+    )
+    close_times = (
+        (None, time(16, 30)),
```
[the remark this answers is not in the document]
Comment (alberto-exabel) [reply trading_calendars/exchange_calendar_xosl.py]: It is currently regular time. But the closing time has been moved between 16:20 and 17:20 several in recent history.

[the remark this answers is not in the document]
Comment (alberto-exabel) [reply trading_calendars/exchange_calendar_xosl.py]: It is the regular trading time at the moment. But the closing time has been moved between 16:20 and 17:20 several times.

[the remark this answers is not in the document]
Comment (dmichalowicz) [reply trading_calendars/exchange_calendar_xosl.py]: Hmm, if it's changed closing times in the past it'd be good to encode that, I'll see if I can find that history.

File: tests/test_xcse_calendar.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+from unittest import TestCase
+import pandas as pd
+
+from .test_trading_calendar import ExchangeCalendarTestBase
+from trading_calendars.exchange_calendar_xcse import XCSEExchangeCalendar
+
+
+class XCSECalendarTestBase(ExchangeCalendarTestBase, TestCase):
```
[the remark this answers is not in the document]
Comment (dmichalowicz) [reply tests/test_xcse_calendar.py]: yup

File: trading_calendars/exchange_calendar_xsto.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    month=1,
+    day=1,
+    offset=[Easter(), Day(38)],
+)
+AscensionDay = ascension_day()
+WhitMonday = whit_monday(end_date='2005')
+
+NationalDay = Holiday('Sweden National Day', month=6, day=6)
+
+MidsummerEve = Holiday(
```
[the remark this answers is not in the document]
Comment (dmichalowicz) [reply trading_calendars/exchange_calendar_xsto.py]: yup

File: tests/test_xsto_calendar.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+                pd.Timestamp('2018-04-30 13:00', tz='Europe/Stockholm'),
+            ),
+            # Day before Ascension Day.
+            (
+                pd.Timestamp('2018-05-09', tz='UTC'),
+                pd.Timestamp('2018-05-09 13:00', tz='Europe/Stockholm'),
+            ),
+            # Halloween.
+            (
+                pd.Timestamp('2018-10-31', tz='UTC'),
```
[the remark this answers is not in the document]
Comment (dmichalowicz) [reply tests/test_xsto_calendar.py]: Good question, according to http://www.nasdaqomxnordic.com/vidskiptatimi and the link you posted I venture to guess that it's actually the Friday before November 5th.

**Pull request merged**